addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -286,58 +302,185 @@ else if ( topPath.endsWith( "/" ) )
     @Override
     public Transfer generateGroupFileContent( final Group group, final List<ArtifactStore> members, final String path,
                                               final EventMetadata eventMetadata )
-        throws IndyWorkflowException
+            throws IndyWorkflowException
+    {
+        final Metadata md = generateGroupMetadata( group, members, path );
+        if ( md != null )
+        {
+            String toMergePath = path;
+            if ( !path.endsWith( MavenMetadataMerger.METADATA_NAME ) )
+            {
+                toMergePath = normalize( normalize( parentPath( toMergePath ) ), MavenMetadataMerger.METADATA_NAME );
+            }
+
+            final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            Transfer target = null;
+            try
+            {
+                target = fileManager.getTransfer( group, toMergePath );
+                if ( target.exists() )
+                {
+                    target.delete();
+                }
+                new MetadataXpp3Writer().write( baos, md );
+
+                final byte[] merged = baos.toByteArray();
+                if ( merged != null )
+                {
+                    OutputStream fos = null;
+                    try
+                    {
+                        fos = target.openOutputStream( TransferOperation.GENERATE, true, eventMetadata );
+                        fos.write( merged );
+                    }
+                    catch ( final IOException e )
+                    {
+                        throw new IndyWorkflowException( "Failed to write merged metadata to: {}.\nError: {}", e,
+                                                         target, e.getMessage() );
+                    }
+                    finally
+                    {
+                        closeQuietly( fos );
+                    }
+
+                    final List<Transfer> sources =
```

Comment:
Can we store this information in the cache too? We might have to have a nested class or something that combines the Metadata object with the info on where it was generated from, but that should be pretty easy. If we're doing retrieveAllRaw() here, I don't think we're going to save much time. This will hit the filesystem for each member repository, if only to do a File.exists() call...it's all expensive when it's on NFS.

Reply:
I need to think about how to do this.